src/frontend/src/optimizer/plan_node/stream_materialize.rs

```diff
@@ -91,7 +91,7 @@ impl StreamMaterialize {
         let input = required_dist.enforce_if_not_satisfies(input, &Order::any())?;
         let base = PlanBase::derive_stream_plan_base(&input);
         let schema = &base.schema;
-        let pk_indices = &base.logical_pk;
+        let pk_indices = &base.logical_pk.iter().copied().unique().collect_vec();
```

Comment:
Does this include order by columns? If yes, we can't dedupe

Reply:
I think not, the `user_order_by` is order-dependent